File: src/main/java/org/pixeltime/enchantmentsenhance/event/enchantment/Pyromaniac.java
package org.pixeltime.enchantmentsenhance.event.enchantment;

import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageEvent;
import org.bukkit.inventory.ItemStack;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

public class Pyromaniac implements Listener {
    @EventHandler
    public void onDamage(final EntityDamageEvent entityDamageEvent) {
        if (entityDamageEvent.getEntity() instanceof Player) {
            final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantment." + "pyromaniac"));
            final Player player = (Player) entityDamageEvent.getEntity();
            if (entityDamageEvent.getCause() == EntityDamageEvent.DamageCause.FIRE || entityDamageEvent.getCause() == EntityDamageEvent.DamageCause.FIRE_TICK) {
                final int n = (int) (Math.random() * 100.0);
                try {
                    ItemStack[] armorContents = player.getInventory().getArmorContents();
                    for (int length = armorContents.length, i = 0; i < length; ++i) {
                        final ItemStack itemStack = armorContents[i];
                        if (itemStack != null) {
                            if (itemStack.hasItemMeta() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I") && n < SettingsManager.enchant.getInt("pyromaniac.level_I.chance")) {
                                player.setHealth(20.0);
                                player.setFoodLevel(20);
                            }
                            if (itemStack.hasItemMeta() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " II") && n < SettingsManager.enchant.getInt("pyromaniac.level_II.chance")) {
                                player.setHealth(20.0);
                                player.setFoodLevel(20);
                            }
                            if (itemStack.hasItemMeta() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " III") && n < SettingsManager.enchant.getInt("pyromaniac.level_III.chance")) {
                                player.setHealth(20.0);
                                player.setFoodLevel(20);
                            }
                            if (itemStack.hasItemMeta() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " IV") && n < SettingsManager.enchant.getInt("pyromaniac.level_IV.chance")) {
                                player.setHealth(20.0);
                                player.setFoodLevel(20);
                            }
                            if (itemStack.hasItemMeta() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " V") && n < SettingsManager.enchant.getInt("pyromaniac.level_V.chance")) {
                                player.setHealth(20.0);
                                player.setFoodLevel(20);
                            }
                        }
                    }
                } catch (Exception ex) {
                }
            }
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantment/Pyromaniac.kt
/*
 *     Copyright (C) 2017-Present HealPotion
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */

package org.pixeltime.enchantmentsenhance.event.enchantment

import org.bukkit.ChatColor
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.EntityDamageEvent
import org.pixeltime.enchantmentsenhance.manager.IM
import org.pixeltime.enchantmentsenhance.manager.KM
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

class Pyromaniac : Listener {
    @EventHandler
    fun onDamage(entityDamageEvent: EntityDamageEvent) {
        if (entityDamageEvent.entity is Player) {
            val translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantment." + "pyromaniac"))
            val player = entityDamageEvent.entity as Player
            if (entityDamageEvent.cause == EntityDamageEvent.DamageCause.FIRE || entityDamageEvent.cause == EntityDamageEvent.DamageCause.FIRE_TICK) {
                try {
                    val armorContents = player.inventory.armorContents + IM.getAccessorySlots(player)
                    for (itemStack in armorContents) {
                        if (itemStack.hasItemMeta() && itemStack.itemMeta.hasLore()) {
                            val level = KM.getLevel(translateAlternateColorCodes, itemStack.itemMeta.lore)
                            if (level > 0 && (Math.random() * 100.0).toInt() < SettingsManager.enchant.getInt("pyromaniac.$level.chance")) {
                                player.health = 20.0
                                player.foodLevel = 20
                            }
                        }
                    }
                } catch (ex: Exception) {
                }
            }
        }
    }
}
